modules/user_accounts/php/NDB_Form_user_accounts.class.inc

```diff
@@ -829,6 +834,34 @@ class NDB_Form_User_Accounts extends NDB_Form
     }
 
     /**
+    * Query database on UserID to check that password has doesn't
+    * exist and account is still pending approval, to ensure the
+    * users with account activity can't be rejected by setting
+    * their status to pending.
+    *
+    * @param string $id identifier of account to verify
+    *
+    * @return boolean true is Password_hash is null and
+    * Pending_approval is Yes, false otherwsie
+    */
+    function _canRejectAccount($id)
```

Comment:
This doesn't use any internal class variables, so could be made static and reused from the ajax script instead of duplicating the code.